lib/matplotlib/axes/_base.py

```diff
@@ -2874,8 +2874,14 @@ def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
 
         self._process_unit_info(xdata=(left, right))
         if left is not None:
+            if (isinstance(left, float) and
```

Comment:
this is code block is repeater several times throughout the PR.

i recommend factoring it out into it's own method. something like:

```python
...
class _AxesBase(martist.Artist):
    ...
    def _validate_axis_limit(limit):
        if limit is not None:
            if isinstance(limit, float) and
                    (not np.isreal(limit) or not np.isfinite(right)):
                raise ValueError(...)
            return self.convert_xunits(limit)
```

and then all of these blocks can just be, e.g.
```python
right = self._validate_axis_limit(right)
```